test/activerecord/codec_factory_active_record_test.rb

```diff
@@ -6,7 +6,9 @@ class PaquitoCodecFactoryActiveRecordTest < PaquitoTest
   test "MessagePack factory correctly encodes AR::Base objects" do
     shop = Shop.preload(:products, :domain).first
 
-    codec = Paquito::CodecFactory.build([ActiveRecord::Base])
+    codec = Paquito::CodecFactory.build([
+      ActiveRecord::Base, Symbol, Time, DateTime, Date, BigDecimal, ActiveSupport::TimeWithZone,
```

Comment:
I think in "recursive mode" `.build([ActiveRecord::Base])` should append the other necessary types. 

The problem is that it then allows you to use these types even outside of a record.

Reply:
We could have something like "dependent types" , so if you try to call `Paquito::Types.register` with `ActiveRecordCoder` but without the other types it needs, it would raise or at least warn that it needs those types.

Reply:
Hum, I think I like the explicitness of it, this way it's clear you are supporting other types.

Reply:
Ok, I need to think a bit about implementation, I'll come back to this later this week.

Reply:
Alternatively we could keep the "manual nesting". It's unclear what we gain by using `recursive` here.

Reply:
> It's unclear what we gain by using `recursive` here.

We get rid of all of `ActiveRecordPacker`, that seems significant no?

Reply:
It's not that much code, and it allow for better error messages. I'm not 100% against, but just saying the win isn't that clear to me.

Reply:
I quite like that it leverages a feature (`recursive`) that suits the use case very closely. The only downside I see is that the type here has implicit dependencies, but cutting 50 lines of code seems like a win. Agree about error messages but maybe we can find another way to do that 🤔

Reply:
Ok I tried adding a simple implementation of dependent types. I'm not sure if I like it. There are many cases where you don't actually need all the types, but are required to declare them anyway.